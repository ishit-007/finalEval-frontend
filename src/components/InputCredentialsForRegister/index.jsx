/* eslint-disable no-unused-vars */
import React from 'react';
import axios from 'axios';
import './index.css';
import { useNavigate } from 'react-router-dom';

const InputCredentialsForRegister = () => {
  const [userName, setUserName] = React.useState('');
  const [password, setPassword] = React.useState('');

  const registrationHandler = () => {
    axios
      .post('http://localhost:4000/user', {
        userName,
        password,
      })
      .then(resp => {
        navigate('/login');
      })
      .catch(err => {
        alert('User already exists');
      });
  };

  const userNameChangeHandler = event => {
    setUserName(event.target.value);
  };

  const passwordChangeHandler = event => {
    setPassword(event.target.value);
  };

  const navigate = useNavigate();

  return (
    <div className="inputForm">
      <div className="heading">
        <h1>Register to your CMS+ Account</h1>
      </div>
      <div className="emailInput">
        <label htmlFor="">Email</label>
        <input type="email" name="email" id="email" onChange={userNameChangeHandler} />
      </div>
      <div className="passwordInput">
        <label htmlFor="">Password</label>
        <input type="password" name="password" id="password" onChange={passwordChangeHandler} />
      </div>
      <div className="submitBtn">
        <button type="button" onClick={registrationHandler}>
          Register
        </button>
        <a href="">Forgot Password?</a>

        <a
          href=""
          className="other-Option"
          onClick={() => {
            navigate('/login');
          }}>
          Login Instead
        </a>
      </div>
    </div>
  );
};

export default InputCredentialsForRegister;
