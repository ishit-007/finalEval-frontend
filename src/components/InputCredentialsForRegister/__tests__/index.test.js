import InputCredentialsForRegister from '..';
import renderer from 'react-test-renderer';
import React from 'react';

jest.mock('react-router-dom');

describe('InputCredentialsForRegister for Login', () => {
  it('should render the InputCredentialsForRegister component', async () => {
    const tree = renderer.create(<InputCredentialsForRegister />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});