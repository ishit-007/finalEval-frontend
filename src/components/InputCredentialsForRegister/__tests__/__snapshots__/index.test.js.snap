// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCredentialsForRegister for Login should render the InputCredentialsForRegister component 1`] = `
<div
  className="inputForm"
>
  <div
    className="heading"
  >
    <h1>
      Register to your CMS+ Account
    </h1>
  </div>
  <div
    className="emailInput"
  >
    <label
      htmlFor=""
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="passwordInput"
  >
    <label
      htmlFor=""
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="submitBtn"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
    <a
      href=""
    >
      Forgot Password?
    </a>
    <a
      className="other-Option"
      href=""
      onClick={[Function]}
    >
      Login Instead
    </a>
  </div>
</div>
`;
